src/ResourceManager/OperationalInsights/Commands.OperationalInsights/DataSources/NewDataSourceCmdletsPerKind/NewAzureOperationalInsightsWindowsPerformanceObjectDataSourceCommand.cs

```diff
@@ -0,0 +1,62 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.OperationalInsights.Models;
+using Microsoft.Azure.Commands.OperationalInsights.Properties;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.OperationalInsights
+{
+    [Cmdlet(VerbsCommon.New, "AzureRmOperationalInsightsWindowsPerformanceCounterDataSource", SupportsShouldProcess = true, 
+        DefaultParameterSetName = ByWorkspaceName), OutputType(typeof(PSDataSource))]
+    public class NewAzureOperationalInsightsWindowsPerformanceCounterDataSourceCommand : NewAzureOperationalInsightsDataSourceBaseCmdlet
+    {
+        // Base class already take first 4 parameter position.
+        [Parameter(Position = 4, Mandatory = true, ValueFromPipelineByPropertyName = true,
+        HelpMessage = "The name of object name of Linux Performance Counter.")]
+        [ValidateNotNullOrEmpty]
+        public string ObjectName { get; set; }
+
+        [Parameter(Position = 5, Mandatory = true, ValueFromPipelineByPropertyName = true,
+        HelpMessage = "The name of instance name of Linux Performance Counter.")]
+        [ValidateNotNullOrEmpty]
+        public string InstanceName { get; set; }
+
+        [Parameter(Position = 6, Mandatory = true, ValueFromPipelineByPropertyName = true,
+        HelpMessage = "The array of countername for Linux Performance Counter.")]
+        [ValidateNotNullOrEmpty]
+        public string CounterName { get; set; }
+
+        [Parameter(Position = 7, Mandatory = true, ValueFromPipelineByPropertyName = true,
```

Comment:
It is way to excessive to have 7 positional parameters. Please consider having no more than 4